modules/table/src/main/java/org/apache/ignite/internal/table/distributed/gc/GcUpdateHandler.java

```diff
@@ -69,41 +68,74 @@ public PendingComparableValuesTracker<HybridTimestamp, Void> getSafeTimeTracker(
      *
      * @param lowWatermark Low watermark for the vacuum.
      * @param count Count of entries to GC.
+     * @param strict {@code true} if needed to remove the strictly passed {@code count} oldest stale entries, {@code false} if a premature
+     *      exit is allowed when it is not possible to acquire a lock for the {@link RowId}.
      * @return {@code False} if there is no garbage left in the storage.
      */
-    public boolean vacuumBatch(HybridTimestamp lowWatermark, int count) {
+    public boolean vacuumBatch(HybridTimestamp lowWatermark, int count, boolean strict) {
+        if (count <= 0) {
+            return true;
+        }
+
+        IntHolder countHolder = new IntHolder(count);
```

Comment:
It looks like `IntHolder` is always used for counting down. How about renaming it to `CountDown`? It could also have a method like `finished()` that would check that the count reached zero

Reply:
I'd rather leave it as is because it's closer (as api usage) to using **AtomicInteger**.

Reply:
Why is being closer to `AtomicInteger` is an advantage? `AtomicInteger` is a general-purpose util, and `IntHolder` is in our specific package, and only used in just one place, so it could be specialized